### PR TITLE
fix: normalize all timestamp literals to `datetime.datetime`

### LINKED ISSUE #4997: Unable to  transform python value of type '<class 'numpy.datetime64'>' to DuckDB LogicalTypebug:
### What happened?

When using a `numpy.datetime64` to filter I get an error.

### What version of ibis are you using?

3.2.0

### What backend(s) are you using, if any?

duckdb==0.6.0

### Reproducible Example

```python
from pathlib import Path

import duckdb
import ibis
import numpy as np
import pandas as pd
import requests

TRIP_DATA_PATH = "yellow_tripdata_2022-02.parquet"

if not Path(TRIP_DATA_PATH).exists():
    url = "https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2022-02.parquet"
    response = requests.get(url, verify=False)
    with open(TRIP_DATA_PATH, "wb") as f:
        f.write(response.content)

DUCKDB_PATH = "DuckDB.db"

if not Path(DUCKDB_PATH).exists():
    duckdb_con = duckdb.connect(DUCKDB_PATH)
    duckdb_con.execute(f"CREATE TABLE big_df AS SELECT * FROM '{TRIP_DATA_PATH}'")
    duckdb_con.close()

ibis.options.sql.default_limit = None

db = ibis.duckdb.connect(DUCKDB_PATH)

big_table = db.table("big_df")

data = big_table[big_table.tpep_pickup_datetime >= np.datetime64('2022-02-08T06:01:38.761310')]
data.execute()
```

### Relevant log output

```sh
$ python script2.py
C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\postgres\registry.py:164: UserWarning: locale specific date formats (%c, %x, %X) are not yet implemented for Windows
  warnings.warn(
C:\repos\private\hvplot\.venv\lib\site-packages\duckdb_engine\__init__.py:229: DuckDBEngineWarning: 
duckdb-engine doesn't yet support reflection on indices
  warnings.warn(
Traceback (most recent call last):
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\default.py", line 736, in 
do_execute
    cursor.execute(statement, parameters)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\duckdb_engine\__init__.py", line 120, in execute
    self.c.execute(statement, parameters)
duckdb.NotImplementedException: Not implemented Error: Unable to transform python value of type '<class 'numpy.datetime64'>' to DuckDB LogicalType

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\repos\private\hvplot\script2.py", line 31, in <module>
    data.execute()
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\expr\types\core.py", line 291, in execute
    return self._find_backend(use_default=True).execute(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\base\sql\__init__.py", line 191, in execute
    with self._safe_raw_sql(sql, **kwargs) as cursor:
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\base\sql\alchemy\__init__.py", line 124, in _safe_raw_sql
    yield con.execute(*args, **kwargs)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 1380, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\sql\elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 2124, in _handle_dbapi_exception
    util.raise_(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\util\compat.py", line 210, in raise_
    raise exception
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "C:\repos\private\hvplot\.venv\lib\site-packages\sqlalchemy\engine\default.py", line 736, in 
do_execute
    cursor.execute(statement, parameters)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\duckdb_engine\__init__.py", line 120, in execute
    self.c.execute(statement, parameters)
sqlalchemy.exc.NotSupportedError: (duckdb.NotImplementedException) Not implemented Error: Unable to 
transform python value of type '<class 'numpy.datetime64'>' to DuckDB LogicalType
[SQL: SELECT t0."VendorID", t0.tpep_pickup_datetime, t0.tpep_dropoff_datetime, t0.passenger_count, t0.trip_distance, t0."RatecodeID", t0.store_and_fwd_flag, t0."PULocationID", t0."DOLocationID", t0.payment_type, t0.fare_amount, t0.extra, t0.mta_tax, t0.tip_amount, t0.tolls_amount, t0.improvement_surcharge, t0.total_amount, t0.congestion_surcharge, t0.airport_fee
FROM big_df AS t0
WHERE t0.tpep_pickup_datetime >= CAST(? AS TIMESTAMP WITHOUT TIME ZONE)]
[parameters: (numpy.datetime64('2022-02-08T06:01:38.761310'),)]
(Background on this error at: https://sqlalche.me/e/14/tw8g)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional Context

I'm trying to get ibis with the duckdb backend working with hvPlot/ HoloViews. See https://github.com/holoviz/holoviews/issues/5557#issuecomment-1345157033.

I don't know if this is as expected (its not for me :-)), an issue for ibis or an issue for duckdb

### LINKED ISSUE #4998: bug: numpy dtype dtype('<M8[us]') is not supported in the pandas backend
### What happened?

I am trying to improve the support for Ibis (and DuckDB) with hvPlot and Holoviews. Now I am trying to get cross filtering/ linked selections working https://github.com/holoviz/holoviews/issues/5557. But I get the exception `numpy dtype dtype('<M8[us]') is not supported in the pandas backend.`

```python
from pathlib import Path
import duckdb
import holoviews as hv
import ibis
import numpy
import pandas as pd
import requests
from holoviews.util.transform import dim

import hvplot.ibis


TRIP_DATA_PATH = "yellow_tripdata_2022-02.parquet"
COLOR_MAP = ["#4E79A7", "#F28E2B", "#E15759"]

if not Path(TRIP_DATA_PATH).exists():
    url = "https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2022-02.parquet"
    response = requests.get(url, verify=False)
    with open(TRIP_DATA_PATH, "wb") as f:
        f.write(response.content)

DUCKDB_PATH = "DuckDB.db"

if not Path(DUCKDB_PATH).exists():
    duckdb_con = duckdb.connect(DUCKDB_PATH)
    duckdb_con.execute(f"CREATE TABLE big_df AS SELECT * FROM '{TRIP_DATA_PATH}'")
    duckdb_con.close()

ibis.options.sql.default_limit = None

db = ibis.duckdb.connect(DUCKDB_PATH)

big_table = db.table("big_df")
big_table = big_table.filter(big_table.trip_distance<=25).filter(big_table.tpep_pickup_datetime>=pd.Timestamp('2022-02-01')).filter(big_table.tpep_pickup_datetime<pd.Timestamp('2022-03-01'))

dataset = hv.Dataset(big_table)
selection_expr = ((dim('tpep_pickup_datetime')>=numpy.datetime64('2022-02-11T05:18:56.808112'))&(dim('tpep_pickup_datetime')<=numpy.datetime64('2022-02-18T11:20:35.569423')))&((dim('trip_distance')>=6.024007113218731)&(dim('trip_distance')<=16.49822169531713))
mask = selection_expr.apply(dataset, compute=False, keep_index=True, strict=False)
```

If I add `datetime64[us]` as shown below there is no exception.

image

### What version of ibis are you using?

3.20

### What backend(s) are you using, if any?

DuckDB==0.6.0

### Relevant log output

```sh
$ python script.py
C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\postgres\registry.py:164: UserWarning: locale specific date formats (%c, %x, %X) are not yet implemented for Windows
  warnings.warn(
C:\repos\private\hvplot\.venv\lib\site-packages\duckdb_engine\__init__.py:229: DuckDBEngineWarning: duckdb-engine doesn't yet support reflection on indices
  warnings.warn(
Traceback (most recent call last):
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\pandas\client.py", line 102, in from_numpy_dtype
    return _numpy_dtypes[value]
KeyError: dtype('<M8[us]')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\repos\private\tmp\holoviews\script.py", line 38, in <module>
    mask = selection_expr.apply(dataset, compute=False, keep_index=True, strict=False)
  File "C:\repos\private\tmp\holoviews\holoviews\util\transform.py", line 773, in apply
    data = self._apply_fn(dataset, data, fn, fn_name, args,
  File "C:\repos\private\tmp\holoviews\holoviews\util\transform.py", line 673, in _apply_fn
    data = fn(*args, **kwargs)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\expr\types\generic.py", line 527, in __ge__
    return _binop(ops.GreaterEqual, self, rlz.any(other))
  File "C:\repos\private\hvplot\.venv\lib\site-packages\toolz\functoolz.py", line 304, in __call__
    return self._partial(*args, **kwargs)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\expr\rules.py", line 130, in value
    arg = ir.literal(arg)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\expr\types\generic.py", line 1005, in literal
    inferred_dtype = dt.infer(value)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\multipledispatch\dispatcher.py", line 278, in __call__
    return func(*args, **kwargs)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\pandas\client.py", line 128, in infer_numpy_scalar
    return dt.dtype(value.dtype)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\multipledispatch\dispatcher.py", line 278, in __call__
    return func(*args, **kwargs)
  File "C:\repos\private\hvplot\.venv\lib\site-packages\ibis\backends\pandas\client.py", line 104, in from_numpy_dtype
    raise TypeError(
TypeError: numpy dtype dtype('<M8[us]') is not supported in the pandas backen
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
This PR normalizes all user input timestamp literal values to be `datetime.datetime`.

The purpose of the PR is to address #4997 as well as to reduce the amount of duplicated code that was already doing this.

I chose to use `datetime.datetime` instead of `pd.Timestamp` because more backends support `datetime` out of the box,
and also because `Timestamp` can't handle values that don't fit into a `datetime64[ns]`.

To do this and to keep our existing support for timezones I had to introduce a direct dependency on `pytz`. We already transitively included this dependency.

Note that this is only for literal values and is unrelated to type conversion of columnar expressions/values.

Closes #4997.
Closes #4998.

### CODE REVIEW

**Review by jcrist — APPROVED** [review]
LGTM

**Comment by cpcloud** [comment]
I'll squash this.

File: ibis/expr/datatypes/value.py [hunk 1]
```diff
@@ -312,8 +316,47 @@ def normalize(typ, value):
             elif typ.is_multipolygon():
                 return tuple(normalize(dt.polygon, item) for item in value)
         return _WellKnownText(value.wkt)
+    elif (is_timestamp := typ.is_timestamp()) or typ.is_date():
+        import pandas as pd
```
Comment (kszucs) [inline ibis/expr/datatypes/value.py:320]: We may want to use lazy dispatch here in the future.

**Review by kszucs — APPROVED** [review]

File: ibis/expr/datatypes/value.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/datatypes/value.py:320]: Makes sense!

**Pull request merged**